hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/impl/connector/kafka/StreamKafkaP.java

```diff
@@ -271,43 +266,16 @@ public void restoreFromSnapshot(@Nonnull Object key, @Nonnull Object value) {
         }
     }
 
-    public static class MetaSupplier<K, V, T> implements ProcessorMetaSupplier {
-
-        private final Properties properties;
-        private final List<String> topics;
-        private final DistributedFunction<ConsumerRecord<K, V>, T> projectionFn;
-        private final WatermarkGenerationParams<? super T> wmGenParams;
-        private int totalParallelism;
-
-        public MetaSupplier(
-                @Nonnull Properties properties,
-                @Nonnull List<String> topics,
-                @Nonnull DistributedFunction<ConsumerRecord<K, V>, T> projectionFn,
-                @Nonnull WatermarkGenerationParams<? super T> wmGenParams) {
-            this.properties = new Properties();
-            this.properties.putAll(properties);
-            this.topics = topics;
-            this.projectionFn = projectionFn;
-            this.wmGenParams = wmGenParams;
-        }
-
-        @Override
-        public int preferredLocalParallelism() {
-            return 2;
-        }
-
-        @Override
-        public void init(@Nonnull Context context) {
-            totalParallelism = context.totalParallelism();
-        }
-
-        @Nonnull
-        @Override
-        public Function<Address, ProcessorSupplier> get(@Nonnull List<Address> addresses) {
-            return address -> new CloseableProcessorSupplier<>(
-                    () -> new StreamKafkaP<>(properties, topics, projectionFn, totalParallelism, wmGenParams)
-            );
-        }
+    @Nonnull
+    public static <K, V, T> ProcessorSupplier processorSupplier(
+            @Nonnull Properties properties,
+            @Nonnull List<String> topics,
+            @Nonnull DistributedFunction<ConsumerRecord<K, V>, T> projectionFn,
+            @Nonnull WatermarkGenerationParams<T> wmGenParams
+    ) {
+        return new CloseableProcessorSupplier<StreamKafkaP>(() -> new StreamKafkaP<>(
```

Comment:
We've lost the preferred local parallelism of 2 here.
And the explicit type parameter is not needed.

Reply:
I've moved the preferred parallelism to `KafkaProcessors.streamKafkaP`. Will fix the explicit type params.

Reply:
ok, missed that